src/lib/FlightTasks/tasks/FlightTaskAutoMapper.cpp

```diff
@@ -0,0 +1,171 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file FlightAutoLine.cpp
+ */
+
+#include "FlightTaskAutoMapper.hpp"
+#include <mathlib/mathlib.h>
+
+using namespace matrix;
+
+#define SIGMA_SINGLE_OP			0.000001f
+#define SIGMA_NORM			0.001f
+
+bool FlightTaskAutoMapper::activate()
+{
+	bool ret = FlightTaskAuto::activate();
+	_reset();
+	return ret;
+}
+
+bool FlightTaskAutoMapper::update()
+{
+	// always reset constraints because they might change depending on the type
+	_setDefaultConstraints();
+
+	_updateAltitudeAboveGround();
+
+	bool follow_line = _type == WaypointType::loiter || _type == WaypointType::position;
+	bool follow_line_prev = _type_previous == WaypointType::loiter || _type_previous == WaypointType::position;
+
+	// 1st time that vehicle starts to follow line. Reset all setpoints to current vehicle state.
+	if (follow_line && !follow_line_prev) {
+		_reset();
+	}
+
+	// The only time a thrust set-point is sent out is during
+	// idle. Hence, reset thrust set-point to NAN in case the
+	// vehicle exits idle.
+
+	if (_type_previous == WaypointType::idle) {
+		_thrust_setpoint = Vector3f(NAN, NAN, NAN);
+	}
+
+	if (_type == WaypointType::idle) {
+		_generateIdleSetpoints();
+
+	} else if (_type == WaypointType::land) {
+		_generateLandSetpoints();
+
+	} else if (follow_line) {
+		_generateSetpoints();
+
+	} else if (_type == WaypointType::takeoff) {
+		_generateTakeoffSetpoints();
+
+	} else if (_type == WaypointType::velocity) {
+		_generateVelocitySetpoints();
+	}
+
+	// update previous type
+	_type_previous = _type;
+
+	return true;
+}
+
+void FlightTaskAutoMapper::_reset()
+{
+	// Set setpoints equal current state.
+	_velocity_setpoint = _velocity;
+	_position_setpoint = _position;
+	_yaw_setpoint = _yaw;
+	_destination = _position;
+	_origin = _position;
+	_speed_at_target = 0.0f;
+}
+
+void FlightTaskAutoMapper::_generateIdleSetpoints()
+{
+	// Send zero thrust setpoint */
+	_position_setpoint = Vector3f(NAN, NAN, NAN); // Don't require any position/velocity setpoints
+	_velocity_setpoint = Vector3f(NAN, NAN, NAN);
+	_thrust_setpoint.zero();
+}
+
+void FlightTaskAutoMapper::_generateLandSetpoints()
+{
+	// Keep xy-position and go down with landspeed. */
```

Comment:
I know that it was me who added these comments...
I noticed it now during review that `*/` can be removed